app/src/main/java/io/stormbird/wallet/service/KeyService.java

```diff
@@ -0,0 +1,1412 @@
+            }
+        }
+        return false;
+    }
+
+    static byte[] readBytesFromStream(InputStream in)
+    {
+        // this dynamically extends to take the bytes you read
+        ByteArrayOutputStream byteBuffer = new ByteArrayOutputStream();
+        // this is storage overwritten on each iteration with bytes
+        int bufferSize = 1024;
+        byte[] buffer = new byte[bufferSize];
+        // we need to know how may bytes were read to write them to the byteBuffer
+        int len;
+        try
+        {
+            while ((len = in.read(buffer)) != -1)
+            {
+                byteBuffer.write(buffer, 0, len);
+            }
+        }
+        catch (IOException e)
+        {
+            e.printStackTrace();
+        }
+        finally
+        {
+            try
+            {
+                byteBuffer.close();
+            }
+            catch (IOException e)
+            {
+                e.printStackTrace();
+            }
+            if (in != null)
+            {
+                try
+                {
+                    in.close();
+                }
+                catch (IOException e)
+                {
+                    e.printStackTrace();
+                }
+            }
+        }
+        // and then we can return your byte array.
+        return byteBuffer.toByteArray();
+    }
+
+
+    private synchronized void deleteKey(KeyStore keyStore, String keyAddr)
+    {
+        try
+        {
+            String encryptedBytesPath = getEncryptedFilePath(keyAddr);
+            if (encryptedBytesPath != null && new File(encryptedBytesPath).exists())  new File(encryptedBytesPath).delete();
+            File iv = new File(getFilePath(context, keyAddr + "iv"));
+            if (iv.exists())
+                iv.delete();
+            if (keyStore != null && keyStore.containsAlias(keyAddr))
+                keyStore.deleteEntry(keyAddr);
+        }
+        catch (KeyStoreException e)
+        {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Delete a non-auth key
+     * note that the iv and keystore key will have been replaced by the newly minted auth key
+     *
+     * @param address
+     */
+    private void deleteNonAuthKeyEncryptedKeyBytes(String address)
+    {
+        File nonAuthKey = new File(getFilePath(context, address + NO_AUTH_LABEL + HDKEY_LABEL));
+        File nonAuthKeystore = new File(getFilePath(context, address + NO_AUTH_LABEL + KEYSTORE_LABEL));
+        File legacyKey = new File(getFilePath(context, address));
+        if (nonAuthKey.exists()) nonAuthKey.delete();
+        if (nonAuthKeystore.exists()) nonAuthKeystore.delete();
+        if (legacyKey.exists()) legacyKey.delete();
+    }
+
+    /**
+     * Delete all traces of the key in Android keystore, encrypted bytes and iv file in private data area
+     * @param keyAddress
+     */
+    public synchronized void deleteKeyCompletely(String keyAddress)
```

Comment:
Same high level comment as for `deleteKey`. Delete as much as possible rather than "aborting" if a preceding delete operation fails.